src/PowerShellEditorServices/Language/TokenOperations.cs

```diff
@@ -0,0 +1,270 @@
+            foldableRegions.AddRange(
+                MatchCustomCommentRegionTokenElements(tokens, "region")
+            );
+
+            // Find blocks of line comments # comment1\n# comment2\n...
+            foldableRegions.AddRange(
+                MatchBlockCommentTokenElement(tokens, "comment")
+            );
+
+            // Find comments regions <# -> #>
+            foldableRegions.AddRange(
+                MatchTokenElement(tokens, TokenKind.Comment, "comment")
+            );
+
+            // Remove any null entries. Nulls appear if the folding reference is invalid
+            // or missing
+            foldableRegions.RemoveAll(item => item == null);
+
+            // Sort the FoldingReferences, starting at the top of the document,
+            // and ensure that, in the case of multiple ranges starting the same line,
+            // that the largest range (i.e. most number of lines spanned) is sorted
+            // first. This is needed to detect duplicate regions. The first in the list
+            // will be used and subsequent duplicates ignored.
+            foldableRegions.Sort();
+
+            // It's possible to have duplicate or overlapping ranges, that is, regions which have the same starting
+            // line number as the previous region. Therefore only emit ranges which have a different starting line
+            // than the previous range.
+            foldableRegions.RemoveAll( (FoldingReference item) => {
+                // Note - I'm not happy with searching here, but as the RemoveAll
+                // doesn't expose the index in the List, we need to calculate it. Fortunately the
+                // list is sorted at this point, so we can use BinarySearch.
+                int index = foldableRegions.BinarySearch(item);
+                if (index == 0) { return false; }
+                return (item.startLine == foldableRegions[index - 1].startLine);
+            });
+
+            // Some editors have different folding UI, sometimes the lastline should be displayed
+            // If we do want to show the last line, just change the region to be one line less
+            if (ShowLastLine) {
+                foldableRegions.ForEach( item => { item.endLine--; });
+            }
+
+            return foldableRegions.ToArray();
+        }
+
+        /// <summary>
+        /// Creates an instance of a FoldingReference object from a start and end langauge Token
+        /// Returns null if the line range is invalid
+        /// </summary>
+        static private FoldingReference CreateFoldingReference(
+            Token startToken,
+            Token endToken,
+            string matchKind)
+        {
+            if (endToken.Extent.EndLineNumber == startToken.Extent.StartLineNumber) { return null; }
+            return new FoldingReference(
+                startToken.Extent.StartLineNumber - 1,   // Extents are base 1, but LSP is base 0
```

Comment:
Seems like one comment here would be enough.  :-)

Reply:
Will fix.

Reply:
Fixed